### PR TITLE
Migrate holonix repo

### PR DESCRIPTION
**Pull request by DavHau**
### Summary
Migrates all of holonix into the holochain repo. Holonix is simply copied to `./holonix`.
This change will allow us to work more efficient when doing changes on the nix logic and infra.

Notes:
  - the `holonix` input is removed from `niv`, as holonix is now local.
  - `./holonix/nix/sources.{json,nix}` is now symlinked to `./nix/sources.{json,nix}` (later to be repalced by flake inputs)
  - `holochain-nixpkgs` entry is added via niv. Earlier it was managed by holonix' niv, but that doesn't exist anymore
  - holonix' `workflows` are copied over and modified to operate inside `./holonix`.

### TODO:
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

**Comment by steveeJ** [comment]
> ** holonix integration test / tests (macos-x86_64, macos-latest) (pull_request) ** Failing after 13m

this one fails because it pulls in sysinfo-0.15, while the macos-aarch64 does not do that. how can they have different crate dependencies?

**Review by steveeJ — COMMENTED** [review]
> ** holonix integration test / tests (macos-x86_64, macos-latest) (pull_request) ** Failing after 13m

this one fails because it pulls in sysinfo-0.15, while the macos-aarch64 does not do that. how can they have different crate dependencies?

**Comment by DavHau** [comment]
They al pull sysinfo 0.15, because the launcher still locks an old version of holochain, that has this dependency.
As to why it doesn't fail on the aarch64-darwin, I don't know exactly why. I have the suspicion that this error isn't entirely reproducible. I have seen it fail differently on different machines.

Anyways, the solution should be to get rid of sysinfo by updating the launcher. Just this here seems to block it: https://github.com/holochain/launcher/pull/123

File: nix/modules/writers.nix [hunk 1]
```diff
@@ -0,0 +1,81 @@
+{
```
Comment (steveeJ) [inline nix/modules/writers.nix:1]: could this live somewhere in a library repo that is maintained by the nix-community? i'm hesitant to take it on as i also don't fully understand it as of now, but it seems to reimplement some of `nix-shell --pure`.

**Review by steveeJ — COMMENTED** [review]
nice, thanks!

i would like to move forward with this approach!

i've got a few questions still before i'm comfortable merging.

File: .github/actions/update/action.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      run: |
+        git config --global user.email "hrabot@holochain.org"
+        git config --global user.name "Holochain Release Automation"
+    - name: Set NIX_PATH
+      shell: bash
+      run: "echo NIX_PATH=nixpkgs=$(nix eval --impure --raw --expr '(import (import ./nix/sources.nix {}).holochain-nixpkgs {}).pkgs.path') >> $GITHUB_ENV"
+    - name: update holochain-nixpkgs
+      shell: bash
+      run: |
+        cd ./holonix
```
Comment (steveeJ) [inline .github/actions/update/action.yml:51]:
```suggestion
```

**Pull request merged**